src/applications/personalization/profile360/vet360/actions/transactions.js

```diff
@@ -111,186 +97,41 @@ export function refreshTransaction(transaction, analyticsSectionName) {
-
-      let fieldData = nextFieldValue;
-
-      switch (fieldType) {
-        case 'phone':
-          fieldData = createPhoneObject(nextFieldValue, fieldName);
-          break;
-        case 'email':
-          break;
-        case 'address':
-          fieldData = createAddressObject(nextFieldValue, fieldName);
-          break;
-        default:
-      }
-
-      const options = {
-        body: JSON.stringify(fieldData),
-        method,
-        headers: {
-          'Content-Type': 'application/json'
-        }
-      };
-
-      try {
-        dispatch({
-          type: VET360_TRANSACTION_REQUESTED,
-          fieldName,
-          method
-        });
-
-        const transaction = isVet360Configured() ? await apiRequest(apiRoute, options) : await localVet360.createTransaction();
-
-        recordProfileTransaction('profile-transaction', fieldName);
-
-        dispatch({
-          type: VET360_TRANSACTION_REQUEST_SUCCEEDED,
-          fieldName,
-          transaction
-        });
-      } catch (error) {
-        dispatch({
-          type: VET360_TRANSACTION_REQUEST_FAILED,
-          error,
-          fieldName
-        });
+export function createTransaction(route, method, fieldName, payload, analyticsSectionName) {
+  return async (dispatch) => {
+    const options = {
+      body: JSON.stringify(payload),
+      method,
+      headers: {
+        'Content-Type': 'application/json'
       }
     };
-  };
-}
-
-function deleteVet360Field(apiRoute, fieldName, fieldType) {
-  return () => {
-    return async (dispatch, getState) => {
-      const currentState = getState();
-      const currentFieldValue = currentState.user.profile.vet360[fieldName];
-      let fieldData = currentFieldValue;
-
-      switch (fieldType) {
-        case 'phone':
-          fieldData = createPhoneObject(currentFieldValue, fieldName);
-          break;
-        case 'email':
-          break;
-        case 'address':
-          fieldData = createAddressObject(currentFieldValue, fieldName);
-          break;
-        default:
-      }
 
-      const options = {
-        body: JSON.stringify(fieldData),
-        method: 'DELETE',
-        headers: {
-          'Content-Type': 'application/json'
-        }
-      };
-
-      try {
-        dispatch({
-          type: VET360_TRANSACTION_REQUESTED,
-          fieldName,
-          method: 'DELETE'
-        });
+    try {
+      dispatch({
+        type: VET360_TRANSACTION_REQUESTED,
+        fieldName,
+        method
+      });
 
-        const transaction = isVet360Configured() ? await apiRequest(apiRoute, options) : await localVet360.createTransaction();
+      const transaction = isVet360Configured() ? await apiRequest(route, options) : await localVet360.createTransaction();
 
-        recordProfileTransaction('profile-deleted', fieldName);
+      recordEvent({
+        event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
+        'profile-section': analyticsSectionName
+      });
 
```

Comment:
We previously had a helper function here, `recordProfileTransaction`, that internally was doing a safety check on the action and section names but I think as of now they are always defined so I removed that.